src/System.Private.ServiceModel/src/System/ServiceModel/Channels/WindowsStreamSecurityUpgradeProvider.cs

```diff
@@ -0,0 +1,425 @@
+                    timeoutHelper.GetCancellationToken());
+
+                _impersonationLevel = impersonationLevelWrapper.Value;
+                _allowNtlm = allowNtlmWrapper; 
+
+                return; 
+            }
+
+            internal override void Open(TimeSpan timeout)
+            {
+                OpenAsync(timeout).GetAwaiter();
+            }
+            
+            internal override void Close(TimeSpan timeout)
+            {
+                TimeoutHelper timeoutHelper = new TimeoutHelper(timeout);
+                base.Close(timeoutHelper.RemainingTime());
+                SecurityUtils.CloseTokenProviderIfRequired(_clientTokenProvider, timeoutHelper.RemainingTime());
+            }
+
+            static SecurityMessageProperty CreateServerSecurity(NegotiateStream negotiateStream)
+            {
+                GenericIdentity remoteIdentity = (GenericIdentity)negotiateStream.RemoteIdentity;
+                string principalName = remoteIdentity.Name;
+                if ((principalName != null) && (principalName.Length > 0))
+                {
+                    ReadOnlyCollection<IAuthorizationPolicy> authorizationPolicies = SecurityUtils.CreatePrincipalNameAuthorizationPolicies(principalName);
+                    SecurityMessageProperty result = new SecurityMessageProperty();
+                    result.TransportToken = new SecurityTokenSpecification(null, authorizationPolicies);
+                    result.ServiceSecurityContext = new ServiceSecurityContext(authorizationPolicies);
+                    return result;
+                }
+                else
+                {
+                    return null;
+                }
+            }
+
+            protected override Stream OnInitiateUpgrade(Stream stream, out SecurityMessageProperty remoteSecurity)
+            {
+                OutWrapper<SecurityMessageProperty> remoteSecurityOut = new OutWrapper<SecurityMessageProperty>(); 
+                
+                var retVal = OnInitiateUpgradeAsync(stream, remoteSecurityOut).GetAwaiter().GetResult();
+                remoteSecurity = remoteSecurityOut.Value;
+
+                return retVal;
+            }
+
+            protected override Task<Stream> OnInitiateUpgradeAsync(Stream stream, OutWrapper<SecurityMessageProperty> remoteSecurity)
+            {
+                NegotiateStream negotiateStream;
+                string targetName;
+                EndpointIdentity identity;
+
+                if (TD.WindowsStreamSecurityOnInitiateUpgradeIsEnabled())
+                {
+                    TD.WindowsStreamSecurityOnInitiateUpgrade();
+                }
+
+                // prepare
+                InitiateUpgradePrepare(stream, out negotiateStream, out targetName, out identity);
+
+                // authenticate
+                try
+                {
+                    negotiateStream.AuthenticateAsClient(_credential, targetName, _parent.ProtectionLevel, _impersonationLevel);
```

Comment:
The NCL team are removing the sync versions of this method in this class and in SslStream. You need to switch to AuthenticateAsClientAsync method.